src/video_core/renderer_vulkan/vk_texture_cache.cpp

```diff
@@ -806,6 +822,30 @@ TextureCacheRuntime::TextureCacheRuntime(const Device& device_, Scheduler& sched
         astc_decoder_pass.emplace(device, scheduler, descriptor_pool, staging_buffer_pool,
                                   compute_pass_descriptor_queue, memory_allocator);
     }
+    if (!device.IsKhrImageFormatListSupported()) {
+        return;
+    }
+    for (size_t index_a = 0; index_a < view_formats.size(); index_a++) {
+        const PixelFormat image_format = static_cast<PixelFormat>(index_a);
+        const SurfaceType type_a = VideoCore::Surface::GetFormatType(image_format);
+        const auto format_info =
+            MaxwellToVK::SurfaceFormat(device, FormatType::Optimal, true, image_format);
+        view_formats[index_a].push_back(format_info.format);
+        if (type_a != SurfaceType::ColorTexture) {
+            continue;
+        }
+        for (size_t index_b = 0; index_b < view_formats.size(); index_b++) {
+            if (index_b == index_a) {
+                continue;
+            }
+            const PixelFormat view_format = static_cast<PixelFormat>(index_b);
+            if (VideoCore::Surface::IsViewCompatible(image_format, view_format, false, true)) {
+                const auto view_info =
+                    MaxwellToVK::SurfaceFormat(device, FormatType::Optimal, true, view_format);
+                view_formats[index_a].push_back(view_info.format);
+            }
+        }
+    }
```

Comment:
```suggestion
    for (size_t index_a = 0; index_a < VideoCore::Surface::MaxPixelFormat; index_a++) {
        const auto image_format = static_cast<PixelFormat>(index_a);
        const auto type_a = VideoCore::Surface::GetFormatType(image_format);
        if (type_a != SurfaceType::ColorTexture) {
            continue;
        }
        for (size_t index_b = 0; index_b < VideoCore::Surface::MaxPixelFormat; index_b++) {
            const auto view_format = static_cast<PixelFormat>(index_b);
            if (VideoCore::Surface::IsViewCompatible(image_format, view_format, false, true)) {
                const auto view_info =
                    MaxwellToVK::SurfaceFormat(device, FormatType::Optimal, true, view_format);
                view_formats[index_a].push_back(view_info.format);
            }
        }
    }
```
(a bit more concise)